### PR TITLE
[SPARK-20378][CORE][SQL][SS] StreamSinkProvider should provide schema in createSink

### PR DESCRIPTION
**Pull request by ymahajan**
## What changes were proposed in this pull request?

Provided schema in StreamSinkProvider#createSink for custom Sink implementations.

## How was this patch tested?

manual tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by felixcheung** [comment]
should this be `[SS]` other than `[Streaming]`?

**Comment by ymahajan** [comment]
@felixcheung Thanks, I changed it to [SS] in the subject of this PR.

File: sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala [hunk 1]
```diff
@@ -151,6 +151,7 @@ trait StreamSourceProvider {
 @InterfaceStability.Unstable
 trait StreamSinkProvider {
   def createSink(
+      schema: StructType,
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala]: The interface can't just be changed like this. This breaks backwards compatibility ...

**Comment by HyukjinKwon** [comment]
ping @ymahajan

**Pull request closed without merging**